pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -201,14 +197,39 @@ func (c *Cluster) Start() error {
 }
 
 func (c *Cluster) removeExtraMgrs(daemonIDs []string) {
+	options := metav1.ListOptions{LabelSelector: "app=" + AppName}
+	mgrDeployments, err := c.context.Clientset.AppsV1().Deployments(c.clusterInfo.Namespace).List(c.clusterInfo.Context, options)
+	if err != nil {
+		logger.Warningf("failed to check for extra mgrs. %v", err)
+		return
+	}
+	if len(mgrDeployments.Items) == len(daemonIDs) {
```

Comment:
If there are fewer deployments, then the loop below will just not find any to remove. But I don't expect that to happen actually because this is called right after all the expected mgr daemons are reconciled.